espnet2/bin/slu_inference.py

```diff
@@ -0,0 +1,702 @@
+#!/usr/bin/env python3
+import argparse
+import logging
+import sys
+from distutils.version import LooseVersion
+from pathlib import Path
+from typing import Any, List, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import torch
+import torch.quantization
+from typeguard import check_argument_types, check_return_type
+
+from espnet2.asr.transducer.beam_search_transducer import BeamSearchTransducer
+from espnet2.asr.transducer.beam_search_transducer import (
+    ExtendedHypothesis as ExtTransHypothesis,
+)
+from espnet2.asr.transducer.beam_search_transducer import Hypothesis as TransHypothesis
+from espnet2.fileio.datadir_writer import DatadirWriter
+from espnet2.tasks.lm import LMTask
+from espnet2.tasks.slu import SLUTask
+from espnet2.text.build_tokenizer import build_tokenizer
+from espnet2.text.token_id_converter import TokenIDConverter
+from espnet2.torch_utils.device_funcs import to_device
+from espnet2.torch_utils.set_all_random_seed import set_all_random_seed
+from espnet2.utils import config_argparse
+from espnet2.utils.types import str2bool, str2triple_str, str_or_none
+from espnet.nets.batch_beam_search import BatchBeamSearch
+from espnet.nets.batch_beam_search_online_sim import BatchBeamSearchOnlineSim
+from espnet.nets.beam_search import BeamSearch, Hypothesis
+from espnet.nets.pytorch_backend.transformer.subsampling import TooShortUttError
+from espnet.nets.scorer_interface import BatchScorerInterface
+from espnet.nets.scorers.ctc import CTCPrefixScorer
+from espnet.nets.scorers.length_bonus import LengthBonus
+from espnet.utils.cli_utils import get_commandline_args
+
+
+class Speech2Understand:
+    """Speech2Understand class
+
+    Examples:
+        >>> import soundfile
+        >>> speech2understand = Speech2Understand("slu_config.yml", "slu.pth")
+        >>> audio, rate = soundfile.read("speech.wav")
+        >>> speech2understand(audio)
+        [(text, token, token_int, hypothesis object), ...]
```

Comment:
The output looks complicated as an API.
Are all of them necessary?

Reply:
I tried to make it exactly same as Speech2Text https://github.com/espnet/espnet/blob/f274ebed88e3b4820b23ea71fb7f9f6d56706be9/espnet2/bin/asr_inference.py#L47 so that I can use the same inference API as ASR in Hugging face. Let me know if you still think it needs to be simplified.